src/mpi/coll/reduce/reduce.c

```diff
@@ -263,9 +274,20 @@ int MPIR_Reduce_impl(const void *sendbuf, void *recvbuf, int count,
                      MPIR_Comm * comm_ptr, MPIR_Errflag_t * errflag)
 {
     int mpi_errno = MPI_SUCCESS;
+    int pof2;
+
+    pof2 = comm_ptr->pof2;
 
     if (comm_ptr->comm_kind == MPIR_COMM_KIND__INTRACOMM) {
         /* intracommunicator */
+        if (MPIR_CVAR_REDUCE_INTRA_ALGORITHM_CONST_SPACE) {
+            if (count >= pof2)
+                MPIR_Reduce_intra_algo_choice =
```

Comment:
We should leave this value alone here. It is already set in `MPII_Coll_init`.

Reply:
@raffenet There is an assert to check whether count>= pof2 in reduce_scatter new implementation. When const_space implementation is selected through CVAR, this assertion fails. Hence i added this code to use binomial algorithm when count < pof2.